pkg/clientcert/cert_controller.go

```diff
@@ -166,7 +205,47 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
 
 	// reconcile pending csr if exists
 	if len(c.csrName) > 0 {
-		newSecretConfig, err := c.syncCSR()
+		// build a secret data map if the csr is approved
+		newSecretConfig, err := func() (map[string][]byte, error) {
+			// skip if there is no ongoing csr
+			if len(c.csrName) == 0 {
+				return nil, fmt.Errorf("no ongoing csr")
+			}
+
+			csr, err := c.csrControl.get(c.csrName)
+			if err != nil {
+				return nil, err
+			}
+
+			// skip if csr is not approved yet
+			if !c.csrControl.isApproved(csr) {
+				return nil, nil
+			}
+
+			// skip if csr has no certificate in its status yet
+			certData, issued := c.csrControl.isIssued(csr)
```

Comment:
nit: change the current comment: skip if csr is not issued

Reply:
done